uaclient/entitlements/base.py

```diff
@@ -194,16 +197,12 @@ def applicability_status(self) -> "Tuple[ApplicabilityStatus, str]":
 
             if not match:
                 return ApplicabilityStatus.INAPPLICABLE, invalid_msg
-            if any(
-                [
-                    int(match.group("major")) < min_kern_major,
-                    int(match.group("minor")) < min_kern_minor,
-                ]
-            ):
+            min_kern = float("{}.{}".format(min_kern_major, min_kern_minor))
+            current_kern = float(
+                "{}.{}".format(match.group("major"), match.group("minor"))
+            )
+            if current_kern < min_kern:
```

Comment:
This won't work for minor versions above 9:

```py
In [3]: float("4.15") < float("4.4")
Out[3]: True
```

Reply:
sonofagun. Right, will correct this in separate PR.